web/modules/custom/dpl_reservations/dpl_reservations.module

```diff
@@ -22,3 +24,57 @@ function dpl_reservations_dpl_react_apps_data(array &$data): void {
     'reservation-list-loan-before' => t('Borrow before @date', [], ['context' => 'Reservation list']),
   ];
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * @param array<mixed> $form
+ *   An associative array containing the structure of the form.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current state of the form.
+ */
+function dpl_reservations_form_dpl_library_agency_list_size_settings_form_alter(array &$form, FormStateInterface $form_state): void {
+  $config = \Drupal::service('config.factory')
+    ->getEditable('dpl_library_agency.list_size_settings');
+
+  $form['reservation_list_size_settings'] = [
+    '#type' => 'fieldset',
+    '#title' => t('Reservation list size settings', [], ['context' => 'List size (settings)']),
+    '#description' => t('The number of items to display in the reservation list.', [], ['context' => 'List size (settings)']),
+    '#tree' => FALSE,
+  ];
+
+  $form['reservation_list_size_settings']['reservation_list_size_desktop'] = [
+    '#type' => 'number',
+    '#title' => t('Reservation list size on desktop', [], ['context' => 'List size (settings)']),
+    '#default_value' => $config->get('reservation_list_size_desktop') ?? DplReservationsSettings::RESERVATION_LIST_SIZE_DESKTOP,
+    '#min' => 1,
+    '#step' => 1,
+  ];
+
+  $form['reservation_list_size_settings']['reservation_list_size_mobile'] = [
+    '#type' => 'number',
+    '#title' => t('Reservation list size on mobile', [], ['context' => 'List size (settings)']),
+    '#default_value' => $config->get('reservation_list_size_mobile') ?? DplReservationsSettings::RESERVATION_LIST_SIZE_MOBILE,
+    '#min' => 1,
+    '#step' => 1,
+  ];
+
+  $form['#submit'][] = 'dpl_reservations_handler_submit';
+}
+
+/**
+ * Custom submit callback.
+ *
+ * @param array<mixed> $form
+ *   An associative array containing the structure of the form.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current state of the form.
+ */
+function dpl_reservations_handler_submit(array &$form, FormStateInterface $form_state): void {
+  \Drupal::service('config.factory')
+    ->getEditable('dpl_library_agency.list_size_settings')
```

Comment:
You should use config service here instead.

Reply:
I tried implementing the use of config service here, but only with limited success.
Config service will not let me use `->set()` function, as the config is immutable.

Do you have another suggestion?